modules/packages/LinearAlgebra.chpl

```diff
@@ -1678,6 +1678,49 @@ module Sparse {
     }
     return Y;
   }
+  
+  /* Sparse Matrix-vector multiplication with distributed support */
+  proc _sparsematvecMult(const ref A: [?Adom] ?eltType, const ref X: [?Xdom] eltType) where isSparseArr(A) {
+    var Y: [Xdom] eltType;
+    var locks$: [LocaleSpace] sync bool;
+    coforall l in Locales do on l {
+      const localDomain = A.localSubdomain();
+      var (low_i, low_j) = localDomain.low;
+      var (high_i, high_j) = localDomain.high;
+      var rowResults: [low_i..high_i] eltType = 0;
+
+      forall (i,j) in localDomain {
+          rowResults[i] += A[i,j] * X[j];
+      }
+      
+      // Writing result
+      // This part assumes 0-based indexing of Adom
```

Comment:
Is this still true? Or is it something forgotten from an earlier version? I don't see anything specific that is based on this assumption but I may be missing something.

Reply:
Maybe it can work with arbitrary indexing after some small edits, but I am afraid my math will be incorrect